### PR TITLE
Add SnapshotConfig

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
@mvines needs SnapshotConfig for grooming snapshot download process

#### Summary of Changes
Extract SnapshotConfig and associated plumbing from https://github.com/solana-labs/solana/pull/5262 into this PR

Fixes #

### CODE REVIEW

File: core/src/bank_forks.rs [hunk 1]
```diff
@@ -14,6 +14,39 @@ use std::path::{Path, PathBuf};
 use std::sync::Arc;
 use std::time::Instant;
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct SnapshotConfig {
+    snapshot_path: PathBuf,
+    snapshot_package_output_path: PathBuf,
+    snapshot_interval: usize,
```
Comment (mvines) [inline core/src/bank_forks.rs]: nit: `snapshot_interval_banks`

File: core/src/bank_forks.rs [hunk 2]
```diff
@@ -14,6 +14,39 @@ use std::path::{Path, PathBuf};
 use std::sync::Arc;
 use std::time::Instant;
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct SnapshotConfig {
```
Comment (sagar-solana) [inline core/src/bank_forks.rs]: should this live in SnapshotUtil ?

File: core/src/bank_forks.rs [hunk 3]
```diff
@@ -14,6 +14,39 @@ use std::path::{Path, PathBuf};
...
+
+    pub fn snapshot_path(&self) -> &Path {
+        self.snapshot_path.as_path()
+    }
+
+    pub fn snapshot_package_output_path(&self) -> &Path {
+        &self.snapshot_package_output_path.as_path()
+    }
+
+    pub fn snapshot_interval(&self) -> usize {
```
Comment (mvines) [inline core/src/bank_forks.rs]: snapshot_interval_banks

File: validator/src/main.rs [hunk 4]
```diff
@@ -159,8 +161,28 @@ fn main() {
...
+                .value_name("OUTPUT_PATHS")
+                .takes_value(true)
+                .requires("snapshot_path")
+                .requires("snapshot_interval_banks")
+                .help("Snapshot package output path"),
+        )
+        .arg(
+            clap::Arg::with_name("snapshot_interval_banks")
+                .long("snapshot-interval-banks")
+                .value_name("OUTPUT_PATHS")
```
Comment (mvines) [inline validator/src/main.rs]: OUTPUT_PATHS -> BANKS

File: validator/src/main.rs [hunk 5]
```diff
@@ -159,8 +161,28 @@ fn main() {
...
+                .takes_value(true)
+                .requires("snapshot_path")
+                .requires("snapshot_package_output_path")
+                .help("Snapshot package output path"),
```
Comment (mvines) [inline validator/src/main.rs]: nit: fix help text

File: core/src/bank_forks.rs (hunk 3 above)
Comment (sagar-solana) [reply core/src/bank_forks.rs]: ^ or snapshot_interval_slots ?

File: core/src/bank_forks.rs (hunk 2 above)
Comment (mvines) [reply core/src/bank_forks.rs]: or just a stand-alone `snapshot_config.rs`?   I don't feel to strongly though

Comment (carllin) [reply core/src/bank_forks.rs]: SnapshotUtil is not introduced in this PR yet xD, we can move it in the main PR.

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Comment by mvines** [comment]
I don't need this busted out anymore, feel free to close and roll this back into #5262

**Pull request closed without merging**